File: golang/1759. 统计同构子字符串的数目.go
func countHomogenous(s string) (res int) {
    const mod int=1e9+7
    pre:=rune(s[0])
    cnt:=0
    for _,c:=range s{
        if c==pre{
            cnt++
        }else{
            res+=(cnt+1)*cnt/2
            cnt=1
            pre=c
        }
    }
    res+=(cnt+1)*cnt/2
    return res%mod
}

File: 刷题/cpp/1759. 统计同构子字符串的数目.cpp
class Solution {
public:
    int countHomogenous(string s) {
        const int MOD=1e9+7;
        char pre=s[0];
        long long cnt=0;
        long long ans=0;
        for(auto ch:s){
            if(ch==pre){
                cnt++;
            }else{
                ans+=((1+cnt)%MOD*cnt%MOD/2)%MOD;
                cnt=1;
                pre=ch;
            }
        }
        ans+=(long long)(1+cnt)*cnt/2;
        return ans%MOD;
    }
};